### PR TITLE
Uri parsing fix

### PR DESCRIPTION
**Pull request by andrewconner**

### CODE REVIEW

File: shoebox/app/com/keepit/common/net/URI.scala [hunk 1]
```diff
@@ -55,15 +55,26 @@ object URI extends Logging {
...
+  def fixDoubleHash(uriString: String): String = {
+    uriString.split("\\#",-1) match {
       case Array(first, rest @ _*) =>
         first + "#" + rest.mkString(encodedHash)
     }
   }
 
+  def encodeSymbols(uriString: String): String = {
+    uriString.map(_ match {
+      case ch if "@$^*()[]{}|".contains(ch) => java.net.URLEncoder.encode(ch.toString, "UTF-8")
```
Comment (gmethvin) [inline shoebox/app/com/keepit/common/net/URI.scala]: You can precompute this as a map and save it as a constant

``` scala
val charToEncoded =
    "@$^()[]{}|".map(c => c -> java.net.URLEncoder.encode(c.toString, "UTF-8")).toMap
```

then do

``` scala
uriString.map(c => charToEncoded.getOrElse(c, c.toString)).mkString
```

File: shoebox/test/com/keepit/common/net/URITest.scala [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import org.specs2.mutable.Specification
+import org.specs2.runner.JUnitRunner
+
+@RunWith(classOf[JUnitRunner])
+class URITest extends Specification {
+
+  "URI" should {
+
+    "parse urls" in {
+      URI.parse("http://google.com/").isDefined === true
```
Comment (gmethvin) [inline shoebox/test/com/keepit/common/net/URITest.scala:13]: FYI, I think `.isDefined === true` is equivalent to `must beSome`.

File: shoebox/app/com/keepit/common/net/URI.scala (hunk 1 above)
Comment (andrewconner) [inline shoebox/app/com/keepit/common/net/URI.scala]: I like that better, thanks. Changed.

**Comment by gmethvin** [comment]
lgtm

File: shoebox/test/com/keepit/common/net/URITest.scala [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package com.keepit.common.net
+
+import org.junit.runner.RunWith
+import org.specs2.mutable.Specification
+import org.specs2.runner.JUnitRunner
+
+@RunWith(classOf[JUnitRunner])
+class URITest extends Specification {
```
Comment (2is10) [inline shoebox/test/com/keepit/common/net/URITest.scala:8]: Thanks for adding a test!  Great start.

**Pull request merged**